cognite/neat/rules/exporter/rules2pydantic_models.py

```diff
@@ -495,21 +495,47 @@ def is_external_id_valid(external_id: str) -> bool:
 
     for edge_one_to_many in self.edges_one_to_many:
         edge_type_id = f"{self.__class__.__name__}.{edge_one_to_many}"
-
-        edges.extend(
-            EdgeApply(
-                space=data_model.space,
-                external_id=f"{self.external_id}-{end_node_id}",
-                type=(data_model.space, edge_type_id),
-                start_node=(data_model.space, self.external_id),
-                end_node=(
-                    data_model.space,
-                    end_node_id,
-                ),
+        if add_class_prefix:
+            edges_list = []
+            dm_view = data_model.views[self.__class__.__name__]
+            if dm_view.properties:
+                for end_node in self.__getattribute__(edge_one_to_many):
+                    if is_external_id_valid(end_node):
+                        mapped_instance = dm_view.properties[edge_one_to_many]
+                        if isinstance(mapped_instance, SingleHopConnectionDefinitionApply):
+                            object_view = mapped_instance.source
+                            if object_view:
+                                object_class_name = object_view.external_id
+                                ext_id_object = add_class_prefix_to_xid(
+                                    class_name=object_class_name, external_id=end_node
+                                )
+                                edge_apply = EdgeApply(
+                                    space=data_model.space,
+                                    external_id=f"{self.external_id}-{ext_id_object}",
+                                    type=(data_model.space, edge_type_id),
+                                    start_node=(data_model.space, self.external_id),
+                                    end_node=(
+                                        data_model.space,
+                                        ext_id_object,
+                                    ),
+                                )
+                                edges_list.append(edge_apply)
+            edges.extend(edges_list)
+        else:
+            edges.extend(
+                EdgeApply(
+                    space=data_model.space,
+                    external_id=f"{self.external_id}-{end_node_id}",
+                    type=(data_model.space, edge_type_id),
+                    start_node=(data_model.space, self.external_id),
+                    end_node=(
+                        data_model.space,
+                        end_node_id,
+                    ),
+                )
+                for end_node_id in self.__getattribute__(edge_one_to_many)
+                if is_external_id_valid(end_node_id)
             )
-            for end_node_id in self.__getattribute__(edge_one_to_many)
-            if is_external_id_valid(end_node_id)
-        )
     return edges
```

Comment:
Suggest rewriting this entire implementation to something like this (you have to verify the implementation)

```python
    class_name = type(self).__name__

    for edge_one_to_many in self.edges_one_to_many:
        edge_type_id = f"{class_name}.{edge_one_to_many}"
        edge_list = []
        for end_node_id in getattr(self, edge_one_to_many):
            if not is_external_id_valid(end_node_id):
                continue
            end_node_external_id = end_node_id
            if add_class_prefix:
                end_node = _get_end_node(data_model[class_name], edge_one_to_many)
                if end_node is None:
                    continue
                end_node_external_id = add_class_prefix_to_xid(end_node, end_node_id)

            edge = EdgeApply(
                space=data_model.space,
                external_id=f"{self.external_id}-{end_node_external_id}",
                type=(data_model.space, edge_type_id),
                start_node=(data_model.space, self.external_id),
                end_node=(data_model.space, end_node_external_id),
            )
            edge_list.append(edge)

    return edges


def _get_end_node(view: ViewApply, edge_one_to_many: str) -> str | None:
    mapping = view.properties[edge_one_to_many]
    if isinstance(view, SingleHopConnectionDefinitionApply) and mapping.source:
        return mapping.source.external_id
    return None
```